cli/tauri.js/src/types/config.ts

```diff
@@ -0,0 +1,43 @@
+// TODO: Clean up types, properly mark which ones are optional
+// May need to have different types for each stage of config generation process
+
+export interface TauriConfig {
```

Comment:
It may be a good idea to consider decomposing the config into different types for each section `build`, `ctx`...  This would better support composing the config by section or passing a single section as a typed object instead of [`any`](https://github.com/tauri-apps/tauri/pull/203/files#diff-f519598f155e7dcc17b4186459b02e32R14).  It would also support extending a single section if needed in the future instead of the entire config.

Reply:
I should note that the `tauri.conf.js` file will soon be a `tauri.conf.json` => although that probably isn't really relevant for these typings.